src/routes/AddCustomDisciplineScreen.tsx

```diff
@@ -78,51 +80,66 @@ interface AddCustomDisciplineScreenPropsType {
 const AddCustomDiscipline = ({ navigation, customDisciplines }: AddCustomDisciplineScreenPropsType): JSX.Element => {
   const [code, setCode] = useState<string>('')
   const [errorMessage, setErrorMessage] = useState<string>('')
+  const [codeToSubmit, setCodeToSubmit] = useState<string>('')
+
+  const { data, loading, error } = useLoadGroupInfo(codeToSubmit)
 
   const showErrorMessage = (message: string): void => {
     setErrorMessage(message)
-    setTimeout(() => {
+    const timeout = setTimeout(() => {
       setErrorMessage('')
     }, 5000)
+    return clearTimeout(timeout)
   }
 
+  useEffect(() => {
+    if (data && !customDisciplines.includes(code)) {
+      AsyncStorage.setCustomDisciplines([...customDisciplines, code])
+        .then(() => {
+          navigation.navigate('Home')
+        })
+        .catch(() => {
+          showErrorMessage(labels.addCustomDiscipline.error.technical)
+        })
+    } else {
+      console.log('no data?')
+    }
+  }, [code, customDisciplines, data, navigation])
+
+  useEffect(() => {
+    if (error && codeToSubmit) {
+      showErrorMessage(labels.addCustomDiscipline.error.wrongCode)
```

Comment:
according to the [hooks FAQ](https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies) showErrorMessage should have a global scope, should itself be a callback or should be injected into this useEffect.

Reply:
@Taggotty not really sure if I got your point, but I added `showErrorMessage` as a dependency and wrapped it in a useCallback.

Reply:
oke with this solution the timeout is not working any more. I just did what Steffen suggested and just use setErrorMessage.